components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/store/UserSessionStore.java

```diff
@@ -404,16 +404,20 @@ public void removeTerminatedSessionRecords(List<String> sessionIdList) {
         }
 
         try (Connection connection = IdentityDatabaseUtil.getDBConnection(true)) {
-
-            deleteSessionDataFromTable(sessionsToRemove, connection, IDN_AUTH_USER_SESSION_MAPPING_TABLE,
-                    SQLQueries.SQL_DELETE_TERMINATED_SESSION_DATA);
-            deleteSessionDataFromTable(sessionsToRemove, connection, IDN_AUTH_SESSION_APP_INFO_TABLE,
-                    SQLQueries.SQL_DELETE_IDN_AUTH_SESSION_APP_INFO);
-            deleteSessionDataFromTable(sessionsToRemove, connection, IDN_AUTH_SESSION_META_DATA_TABLE,
-                    SQLQueries.SQL_DELETE_IDN_AUTH_SESSION_META_DATA);
-            IdentityDatabaseUtil.commitTransaction(connection);
+            try {
+                deleteSessionDataFromTable(sessionsToRemove, connection, IDN_AUTH_USER_SESSION_MAPPING_TABLE,
```

Comment:
Since the deleteSessionDataFromTable repeats three times, can't we move relevant keys and values to map and iterate among them and delete session?